### PR TITLE
Use bio-formats reader for PicoQuant .bin see #275

### PR DESCRIPTION
**Pull request by imunro**

### CODE REVIEW

**Comment by seanwarren** [comment]
@imunro - I've simplified the computation of the intensity images here, could you quickly check that it's ok with the 1xn data please?

File: FLIMfitFrontEnd/Classes/@flim_data_series/compute_tr_data.m [hunk 1]
```diff
@@ -34,15 +34,7 @@
...
-   %{
-    x = linspace(0,3000,100);
-    h = 0;
-    for i=1:obj.n_datasets
-        a = obj.acceptor(:,:,i);
-        h = h + hist(a(:),x);
-    end
-    
-    %}
+
```
Comment (imunro) [inline FLIMfitFrontEnd/Classes/@flim_data_series/compute_tr_data.m]: Simply removing long-commented out historic code.

File: FLIMfitFrontEnd/Classes/@flim_data_series/compute_tr_data.m [hunk 2]
```diff
@@ -156,35 +148,9 @@
...
-        
-        %{
-        in = obj.cur_tr_data(sel,1,:,:);
-        in = sum(in,1);
-        in = sum(in,2);
-        in = squeeze(in);
-        
-        %obj.intensity = in;
-        %}
+        obj.intensity = reshape(in,obj.data_size(3:4)');
```
Comment (imunro) [inline FLIMfitFrontEnd/Classes/@flim_data_series/compute_tr_data.m]: It's not entirely clear to me under what circumstances it need reshaping but this seems pretty safe.

File: FLIMfitFrontEnd/Classes/@flim_data_series/flim_data_series.m [hunk 3]
```diff
@@ -601,14 +601,7 @@ function estimate_g_factor(obj)
...
             s = length(data_size);
-            if s == 3
-                % not sure wht this check is needed
-                % but allow data_size to be 3 for 1xn images
-                if data_size(2) > 1
-                    data_size = [data_size(1) 1 data_size(2:3)];
-                end
-            end
-            obj.data_size = [data_size(:) ; ones(5-s,1)];
+            obj.data_size = [data_size(:); ones(5-s,1)];
```
Comment (imunro) [inline FLIMfitFrontEnd/Classes/@flim_data_series/flim_data_series.m]: Are we confident that this check is no longer needed? I wondered if some "stack of tiffs" format might still need it.

File: FLIMfitFrontEnd/Classes/@flim_data_series/flim_data_series.m [hunk 4]
```diff
@@ -839,18 +832,11 @@ function compute_mask(obj)
...
-                if s(1) == 1 && size(squeezed,1) == s(2)
-                    squeezed = squeezed';
-                end
-                 
-                obj.thresh_mask = obj.intensity >= obj.thresh_min & squeezed < obj.gate_max;
-               
+                max_gate = (max(max(obj.cur_data,[],1),[],2));
+                max_gate = reshape(max_gate,obj.data_size(3:4)');
+                
+                obj.thresh_mask = obj.intensity >= obj.thresh_min & max_gate < obj.gate_max;
```
Comment (imunro) [inline FLIMfitFrontEnd/Classes/@flim_data_series/flim_data_series.m]: This looks good!

File: FLIMfitFrontEnd/Classes/@flim_data_series/compute_tr_data.m (hunk 2 above)
Comment (seanwarren) [inline FLIMfitFrontEnd/Classes/@flim_data_series/compute_tr_data.m]: It's computing the intensity from FLIM data - it's already been summed over the first two dimensions so what's left is a 1x1xnxm image and we want to get a nxm image

File: FLIMfitFrontEnd/Classes/@flim_data_series/flim_data_series.m (hunk 3 above)
Comment (seanwarren) [inline FLIMfitFrontEnd/Classes/@flim_data_series/flim_data_series.m]: This is a leftover from when some of the import functions didn't insert the extra dimension for polarisation data

**Pull request merged**